### PR TITLE
Initial bump-build-number creation

### PR DESCRIPTION
**Pull request by brandonphillips**
Initial creation of bump-build-number step

### CODE REVIEW

**Comment by brandonphillips** [comment]
Please review @dustinvanbuskirk

File: incubating/bump-build-number/codefresh.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+steps:
+  BuildImage:
+    title: Building Docker Image
+    type: build
+    image_name: codefreshplugins/bump-build-number
+  PushingToRegistry:
+    type: push
+    title: Pushing To Registry
+    candidate: ${{BuildImage}}
+    tags: [ "latest", "1.0.0" ]
```
Comment (dustinvanbuskirk) [inline incubating/bump-build-number/codefresh.yml:11]: @brandonphillips please add a conditional around this push for master branch only.

Comment (brandonphillips) [reply incubating/bump-build-number/codefresh.yml:11]: Custom annotation name is there to allow them to track the build number for multiple branches and not just master

Comment (brandonphillips) [reply incubating/bump-build-number/codefresh.yml]: Done - please verify.

**Pull request merged**